src/plugins/intel_cpu/src/nodes/matmul.cpp

```diff
@@ -207,7 +207,11 @@ Node::AttrPtr MatMul::initPrimitiveAttr(const VectorDims &dims) {
 }
 
 Node::AttrPtr MatMul::initPrimitiveAttr() {
-    auto dummyShape = MemoryDescUtils::makeDummyShape(getOutputShapeAtPort(0));
+    auto outputShape = getOutputShapeAtPort(0);
+    for (auto&& node : fusedWith) {
+        outputShape = mergeShapes(outputShape, node->getOutputShapeAtPort(0));
+    }
```

Comment:
It seems we can avoid a bit of code duplication by introducing local function getPropagatedOutputShapeAtPort() or getMinMaxOutputShapeAtPort() (cannot really find a good name)

Reply:
Probably yes, but for such a small amount of duplicated code it is not that necessary to extract it into separate functions.